WalletWasabi.Gui/ViewModels/TextResourceViewModelBase.cs

```diff
@@ -30,28 +34,65 @@ public string Text
 			set => this.RaiseAndSetIfChanged(ref _text, value);
 		}
 
-		public Uri Target { get; }
+		public bool EmptyContent
+		{
+			get => _emptyContent;
+			set => this.RaiseAndSetIfChanged(ref _emptyContent, value);
+		}
+
+		public TextResource TextResource { get; }
 
 		private async Task<string> LoadDocumentAsync(Uri target)
 		{
 			var assetLocator = AvaloniaLocator.Current.GetService<IAssetLoader>();
-
 			using var stream = assetLocator.Open(target);
 			using var reader = new StreamReader(stream);
 			return await reader.ReadToEndAsync();
 		}
 
+		private static async Task<string> LoadDocumentAsync(string path)
+		{
+			using var stream = File.OpenRead(path);
+			using var reader = new StreamReader(stream);
+			return await reader.ReadToEndAsync();
```

Comment:
I would implement it as follows:

```c#
return await File.ReadAllTextAsync(path);
```

Reply:
In fact, in my impl the `await` is not necessary.

Reply:
The function is not necessary at all with file readalltext.